### PR TITLE
types: add omitempty for optional fields

### PR DESCRIPTION
**Pull request by kragniz**
Including lists for consistency with kubernetes core apis.

Co-authored-by: <richard.wall@jetstack.io>

### CODE REVIEW

File: cassandra-operator/pkg/apis/cassandra/v1alpha1/types.go [hunk 1]
```diff
@@ -34,7 +34,8 @@ type Cassandra struct {
 type CassandraSpec struct {
 	// +optional
 	Datacenter *string `json:"datacenter,omitempty"`
-	Racks      []Rack  `json:"racks"`
+	// +optional
+	Racks []Rack `json:"racks,omitempty"`
```
Comment (wallrj) [inline cassandra-operator/pkg/apis/cassandra/v1alpha1/types.go]: I'm  unclear whether `// +optional` is correct here.
It's a required field, but without any validation, it is possible to omit this field.

For example, `Pod.Spec.Containers` is a required list and has neither `// + optional` nor `omitempty`: https://github.com/kubernetes/api/blob/master/core/v1/types.go

Where as `CertificateSigningRequestSpec.Usages` is not marked `// +optional` and yet  marked `omitempty`

Comment (sebbonnet) [reply cassandra-operator/pkg/apis/cassandra/v1alpha1/types.go]: Racks should be required.
We already have validation for it in: https://github.com/sky-uk/cassandra-operator/blob/v0.70.1-alpha/cassandra-operator/pkg/cluster/cluster.go#L152

**Comment by sebbonnet** [comment]
Part of #33

**Comment by kragniz** [comment]
@sebbonnet I've removed rack optionality

**Pull request merged**